src/phasespace/load_mocap/load_mocap.py

```diff
@@ -113,11 +114,33 @@ def get_framerate(self):
     def set_sampling(self, num_samples, mode='uniform'):
         pass
 
-    def set_coordinates(self, markers, new_coords, mode='time-varying'):
-        self._coordinates_mode = mode
-        self._desired_coords = new_coords.squeeze()
-        self._desired_idxs = np.array(markers).squeeze()
-        self._last_transform = np.identity(4)
+    @abstractmethod
+    def register_buffer(self, buffer, **kwargs):
+        """
+        Takes in the reference to a queue-like object and populates it with mocap data processed by the read function 
+        with args and kwargs passed.
+        :param buffer: queue like object
+        :param args: arguments for the read function
+        :param kwargs: keyword arguments for the read function
+        :return: 
+        """
+        pass
+
+    def set_coordinates(self, coordinate_frame_name, markers, new_coords, mode='time-varying'):
```

Comment:
Could we change this to default coordinate_frame_name=None, in which case a single, default frame name will be used? This will keep the API backwards compatible with the previous version (there's a fair amount of other code I have that uses this library). The read() method will also need to be changed to apply the default coordinate transform if no name is specified.